ForMaL/Homeworks/homework1.lean

```diff
@@ -0,0 +1,102 @@
+/-
+# Homework 1: Basics
+References: [MIL] Mathematics in Lean, [Tut] Tutorials project.
+-/
+
+import Mathlib.Tactic
+
+/-
+# Problem  1
+Read the following section of [Tut] and complete the exercises 0013, 0014, 0015. The exercise is to prove the same statement using backwards reasoning, forwards reasoning, and a combination of both.
+-/
+
+/-
+We announce the use of a lemma, and provide proofs after the fact, using
+the `apply` tactic. This is called backward reasoning because we get the conclusion
+first, and provide proofs later. Using `rw` on the goal (rather than on an assumption
+from the local context) is also backward reasoning.
+
+Let's do that using the lemma
+
+  `mul_nonneg {x y : ℝ} (hx : 0 ≤ x) (hy : 0 ≤ y) : 0 ≤ x*y`
```

Comment:
have we explained what the difference between writing `{x y : ℝ}` and `(x y : ℝ)` would mean in the workshop? or should we do that in the homework?

Reply:
We haven't explained in the workshop and I think we should leave it until another time. I've re-written these sections using `#check`. These still have curly braces in the docstring but I think that an important skill (as a learner) is to be able to filter out the parts of the docstring that don't make sense and still extract what the lemma does. My hope is that that is what the students will do here.

Reply:
okok, hopefully